Modules/posixmodule.c

```diff
@@ -2911,6 +2911,11 @@ os_chdir_impl(PyObject *module, path_t *path)
 {
     int result;
 
+    if (PySys_Audit("os.chdir", "(O)",
+                    path->object ? path->object : Py_None) < 0) {
```

Comment:
Pretty sure `path->object` can't be NULL, right? Have you seen a case where this is not true?

I mean, there isn't another way to provide the path if you don't pass an object in...

Reply:
I see there are a few existing cases where we check it like this, but AFAICT none of them should be necessary.

Reply:
For path-like arguments with default values (like `os.listdir` and `os.scandir`), checking `path->object` for `NULL` is necessary, otherwise `SystemError: NULL object passed to Py_BuildValue` could happen. The audit hooks of [`os.listdir`](https://github.com/python/cpython/commit/60419a7e96577cf783b3b45bf3984f9fb0d7ddff#diff-a6f29e907cbb5fffd44d453bcd7b77d5R3763) and [`os.scandir`](https://github.com/python/cpython/commit/60419a7e96577cf783b3b45bf3984f9fb0d7ddff#diff-a6f29e907cbb5fffd44d453bcd7b77d5R13172) are already doing this.
For path-like arguments without default values, it seems that `path->object` could never be `NULL` since the argument is required. The [audit hook in `os.open`](https://github.com/python/cpython/commit/b82e17e626f7b1cd98aada0b1ebb65cb9f8fb184#diff-a6f29e907cbb5fffd44d453bcd7b77d5R8292) does not check `path->object` for `NULL`, and the `os_readlink_impl` function actually assumed that `path->object` is not `NULL` (otherwise the `PyUnicode_Check` call will crash by dereferencing a null pointer).

Reply:
Okay, that sounds good to me. Thanks!